frontend/src/app/page.tsx

```diff
@@ -3,156 +3,200 @@
-                    </InputAdornment>
-                  ),
-                }}
-                sx={{ mb: 4 }}
-              />
-            </form>
-          </Box>
-
-          <Box width="75%" mx="auto" textAlign="center" pb={4}>
-            <Grid container spacing={4}>
-              {['CSCE', 'MATH', 'PHYS', 'ENGL', 'HIST', 'BIOL'].map((course, index) => (
-                <Grid item xs={12} md={4} key={index}>
-                  <div onClick={() => handleCourseClick(course)}>
-                    <Paper 
-                      elevation={3} 
-                      sx={{ 
-                        width: '100%', 
-                        height: 200, 
-                        display: 'flex', 
-                        alignItems: 'center', 
-                        justifyContent: 'center', 
-                        cursor: 'pointer',
-                        backgroundImage: `url(/${course.toLowerCase()}.png)`,
-                        backgroundSize: 'cover',
-                        backgroundPosition: 'center'
-                      }}
-                    >
-                    </Paper>
-                  </div>
-                </Grid>
-              ))}
-            </Grid>
-          </Box>
-        </Stack>
-      </main>
-    </>
-  );
+
+    const handleCourseClick = (course) => {
+        const next = `/dashboard/tutors?major=${encodeURIComponent(course)}`;
+        if (!user) {
+            router.push(`/api/auth/login?next=${encodeURIComponent(next)}`);
+        } else {
+            router.push(next);
+        }
+    };
+
+    const handleSearchSubmit = (event) => {
+        event.preventDefault();
+        if (searchQuery.trim() !== '') {
+            const next = `/dashboard/tutors?query=${encodeURIComponent(searchQuery)}`;
+            if (!user) {
+                router.push(`/api/auth/login?next=${encodeURIComponent(next)}`);
+            } else {
+                router.push(next);
+            }
+        }
+    };
+
+    const [chatOpen, setChatOpen] = useState(false);
+    const [message, setMessage] = useState('');
+    const [conversation, setConversation] = useState<{ content: string; role: string; }[]>([]);
+
+    const handleChatOpen = () => setChatOpen(true);
+    const handleChatClose = () => setChatOpen(false);
+
+    const sendMessage = async () => {
+        setConversation(prevConversation => [...prevConversation, {role: 'user', content: message}]);
+        try {
+            const response = await axios.post(`${process.env.NEXT_PUBLIC_OPENAI_API_URL}`, {
+                model: 'gpt-3.5-turbo',
+                messages: [
+                    {role: 'system', content: 'You are a helpful assistant.'},
+                    {role: 'user', content: message},
+                ],
+                max_tokens: 60,
+            }, {
+                headers: {
+                    'Content-Type': 'application/json',
+                    'Authorization': `Bearer ${process.env.NEXT_PUBLIC_OPENAI_API_KEY}`,
```

Comment:
This is an unsafe way to use the api key. Can be seen by inspecting element while using page